File: src/main/java/com/netsensia/rivalchess/model/helper/CastlingHelper.java
package com.netsensia.rivalchess.model.helper;

import com.netsensia.rivalchess.model.Colour;
import com.netsensia.rivalchess.model.Square;

public class CastlingHelper {

    private CastlingHelper() { }

    public static Square queenRookHome(Colour colour) {
        return Square.fromCoords(0, colour == Colour.WHITE ? 7 : 0);
    }

    public static Square queenKnightHome(Colour colour) {
        return Square.fromCoords(1, colour == Colour.WHITE ? 7 : 0);
    }

    public static Square queenBishopHome(Colour colour) {
        return Square.fromCoords(2, colour == Colour.WHITE ? 7 : 0);
    }

    public static Square queenHome(Colour colour) {
        return Square.fromCoords(3, colour == Colour.WHITE ? 7 : 0);
    }

    public static Square kingHome(Colour colour) {
        return Square.fromCoords(4, colour == Colour.WHITE ? 7 : 0);
    }

    public static Square kingBishopHome(Colour colour) {
        return Square.fromCoords(5, colour == Colour.WHITE ? 7 : 0);
    }

    public static Square kingKnightHome(Colour colour) {
        return Square.fromCoords(6, colour == Colour.WHITE ? 7 : 0);
    }

    public static Square kingRookHome(Colour colour) {
        return Square.fromCoords(7, colour == Colour.WHITE ? 7 : 0);
    }

}

File: src/main/java/com/netsensia/rivalchess/model/helper/CastlingHelper.kt
package com.netsensia.rivalchess.model.helper

import com.netsensia.rivalchess.model.Colour
import com.netsensia.rivalchess.model.Square

object CastlingHelper {
    fun queenRookHome(colour: Colour?): Square {
        return Square.Companion.fromCoords(0, if (colour == Colour.WHITE) 7 else 0)
    }

    fun queenKnightHome(colour: Colour?): Square {
        return Square.Companion.fromCoords(1, if (colour == Colour.WHITE) 7 else 0)
    }

    fun queenBishopHome(colour: Colour?): Square {
        return Square.Companion.fromCoords(2, if (colour == Colour.WHITE) 7 else 0)
    }

    fun queenHome(colour: Colour?): Square {
        return Square.Companion.fromCoords(3, if (colour == Colour.WHITE) 7 else 0)
    }

    fun kingHome(colour: Colour?): Square {
        return Square.Companion.fromCoords(4, if (colour == Colour.WHITE) 7 else 0)
    }

    fun kingBishopHome(colour: Colour?): Square {
        return Square.Companion.fromCoords(5, if (colour == Colour.WHITE) 7 else 0)
    }

    fun kingKnightHome(colour: Colour?): Square {
        return Square.Companion.fromCoords(6, if (colour == Colour.WHITE) 7 else 0)
    }

    fun kingRookHome(colour: Colour?): Square {
        return Square.Companion.fromCoords(7, if (colour == Colour.WHITE) 7 else 0)
    }
}